haystack/components/connectors/openapi_service.py

```diff
@@ -164,28 +166,52 @@ def _authenticate_service(self, openapi_service: OpenAPI, credentials: Optional[
 
     def _invoke_method(self, openapi_service: OpenAPI, method_invocation_descriptor: Dict[str, Any]) -> Any:
         """
-        Invokes the specified method on the OpenAPI service.
+        Invokes the specified method on the OpenAPI service. The method name and arguments are passed in the
+        method_invocation_descriptor.
 
         :param openapi_service: The OpenAPI service instance.
         :type openapi_service: OpenAPI
-        :param method_invocation_descriptor: The method name and arguments.
+        :param method_invocation_descriptor: The method name and arguments to be passed to the method. The payload
+        should contain the method name (key: "name") and the arguments (key: "arguments"). The name is a string, and
+        the arguments are a dictionary of key-value pairs.
         :type method_invocation_descriptor: Dict[str, Any]
         :return: A service JSON response.
         :rtype: Any
         :raises RuntimeError: If the method is not found or invocation fails.
         """
-        name = method_invocation_descriptor["name"]
-        # a bit convoluted, but we need to pass parameters, data, or both to the method
-        # depending on the openapi operation specification, can't use None as a default value
-        method_call_params = {}
-        if (parameters := method_invocation_descriptor["arguments"].get("parameters")) is not None:
-            method_call_params["parameters"] = parameters
-        if (arguments := method_invocation_descriptor["arguments"].get("requestBody")) is not None:
-            method_call_params["data"] = arguments
-
+        name = method_invocation_descriptor.get("name", None)
+        invocation_arguments = copy(method_invocation_descriptor.get("arguments", {}))
+        if not name or not invocation_arguments:
+            raise ValueError(
+                f"Invalid function calling descriptor: {method_invocation_descriptor} . It should contain "
+                f"a method name and arguments."
+            )
+
+        # openapi3 specific method to call the operation, do we have it?
         method_to_call = getattr(openapi_service, f"call_{name}", None)
         if not callable(method_to_call):
             raise RuntimeError(f"Operation {name} not found in OpenAPI specification {openapi_service.info.title}")
 
-        # this will call the underlying service REST API
+        # get the operation reference from the method_to_call
+        operation = method_to_call.operation.__self__
+        operation_dict = operation.raw_element
```

Comment:
I don't get it, why do we need to use the `raw_element` field? `operation` has `parameters` and `requestBody` fields. 🤔

Reply:
Let me check and Ill get back in a minute @silvanocerza

Reply:
Ah right, IIRC correctly I did that so we can then traverse the path easier via dicts.

Reply:
Yes, I confirmed this. It is easier to navigate the tree with dicts, so here we take the dict and then pick up everything safely, because some nodes do not exist always depending on the use case.

Reply:
Ah, now I see why. 
You're mocking the dicts in the tests instead of mocking the object.
That's a bit of a code smell, would be much better if we mock the object properly.